mldsa/poly.c

```diff
@@ -146,20 +156,24 @@ void poly_use_hint(poly *b, const poly *a, const poly *h)
   }
 }
 
+/* Reference: explicitly checks the bound B to be <= (MLDSA_Q - 1) / 8).
+ * This is unnecessary as it's always a compile-time constant.
+ * We instead model it as a precondition.
+ */
 int poly_chknorm(const poly *a, int32_t B)
 {
   unsigned int i;
   int32_t t;
 
-  if (B > (MLDSA_Q - 1) / 8)
-  {
-    return 1;
-  }
-
   /* It is ok to leak which coefficient violates the bound since
      the probability for each coefficient is independent of secret
      data but we must not leak the sign of the centralized representative. */
+
   for (i = 0; i < MLDSA_N; ++i)
+  __loop__(
+    invariant(i <= MLDSA_N)
+    invariant(array_abs_bound(a->coeffs, 0, i, B))
+  )
   {
     /* Absolute value */
     t = a->coeffs[i] >> 31;
```

Comment:
This pattern should be replaced by a CT function: https://github.com/pq-code-package/mldsa-native/issues/152

Reply:
agreed